pkg/occlient/occlient.go

```diff
@@ -1203,3 +1202,44 @@ func (c *Client) GetOneDeploymentConfigFromSelector(selector string) (*appsv1.De
 
 	return &deploymentConfigs[0], nil
 }
+
+// GetOnePodFromSelector returns the Pod  object associated with the given selector.
+// An error is thrown when exactly one Pod is not found.
+func (c *Client) GetOnePodFromSelector(selector string) (*corev1.Pod, error) {
+
+	pods, err := c.kubeClient.CoreV1().Pods(c.namespace).List(metav1.ListOptions{
+		LabelSelector: selector,
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get Pod for the selector: %v", selector)
+	}
+	numDC := len(pods.Items)
+	if numDC == 0 {
+		return nil, fmt.Errorf("no Pod was found for the selector: %v", selector)
+	} else if numDC > 1 {
+		return nil, fmt.Errorf("multiple Pods exist for the selector: %v. Only one must be present", selector)
+	}
+
+	return &pods.Items[0], nil
+}
+
+// SyncPath copies local directory to directory in running Pod.
+func (c *Client) SyncPath(localPath string, targetPodName string, targetPath string) (string, error) {
+	// TODO: do this without using 'oc' binary
+	args := []string{
+		"rsync",
+		localPath,
+		fmt.Sprintf("%s:%s", targetPodName, targetPath),
+		"--exclude", ".git",
```

Comment:
can we also add .swp files to this?

Reply:
We could, but then we are starting to ignore files that are editor specific :-( 
I would like to avoid that.  In future, we should probably add an option so users can set it.

Reply:
+1